grafana-agent: 0.36.2 -> 0.37.1

emilylange opened the pull request:
## Description of changes

https://github.com/grafana/agent/releases/tag/v0.37.0
https://github.com/grafana/agent/blob/v0.37.0/CHANGELOG.md
diff: https://github.com/grafana/agent/compare/v0.36.2...v0.37.0

https://github.com/grafana/agent/releases/tag/v0.37.1
https://github.com/grafana/agent/blob/v0.37.1/CHANGELOG.md
diff: https://github.com/grafana/agent/compare/v0.37.0...v0.37.1

grafana-agent, starting with 0.37.0, requires go >= 1.21.
In nixpkgs (unstable), `buildGoModule` still refers to `buildGo120Module`. Which, as the name suggests, uses go 1.20, and thus does not meet the requirement, so we have to manually override it in `all-packages.nix`.

This override can be dropped very soon again, as commit 2598cb64c13cccc1b380beeaf6b8bba0a472327a titled
`go, buildGoModule, buildGoPackage: default to 1.21` is already in `staging-next` and will land in `master` soon.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

flokli commented on pkgs/top-level/all-packages.nix:
```diff
@@ -26654,7 +26654,9 @@ with pkgs;
 
-  grafana-agent = callPackage ../servers/monitoring/grafana-agent { };
+  grafana-agent = callPackage ../servers/monitoring/grafana-agent {
+    buildGoModule = buildGo121Module;
```
Please describe in the commit message and as a comment here why we're explicitly using `buildGo121Module` instead of `buildGoModule`.

emilylange replied:
Whoops, sorry!

Did amend the commit message just now.
Feel free to click the "Resolve conversation" button, if you feel like this change is good enough :)

flokli replied:
The commit message says `<=`, while it should be `>=`, no?

emilylange replied:
whoops (again).
fixed now :)

emilylange committed:
grafana-agent: 0.36.2 -> 0.37.1

https://github.com/grafana/agent/releases/tag/v0.37.0
https://github.com/grafana/agent/blob/v0.37.0/CHANGELOG.md
diff: https://github.com/grafana/agent/compare/v0.36.2...v0.37.0

https://github.com/grafana/agent/releases/tag/v0.37.1
https://github.com/grafana/agent/blob/v0.37.1/CHANGELOG.md
diff: https://github.com/grafana/agent/compare/v0.37.0...v0.37.1

grafana-agent, starting with 0.37.0, requires go >= 1.21.
In nixpkgs (unstable), `buildGoModule` still refers to
`buildGo120Module`. Which, as the name suggests, uses go 1.20, and thus
does not meet the requirement, so we have to manually override it in
`all-packages.nix`.

This override can be dropped very soon again, as commit
2598cb64c13cccc1b380beeaf6b8bba0a472327a titled
`go, buildGoModule, buildGoPackage: default to 1.21` is already in
`staging-next` and will land in `master` soon.

flokli commented:
Thanks!
